src/NLog/LayoutRenderers/GuidLayoutRenderer.cs

```diff
@@ -59,13 +59,38 @@ public GuidLayoutRenderer()
         public string Format { get; set; }
 
         /// <summary>
+        /// Generate the Guid from the NLog LogEvent (Will be the same for all targets)
+        /// </summary>
+        [DefaultValue(false)]
+        public bool LogEventGuid { get; set; }
```

Comment:
What about "basedOnLogEvent"? Or "GeneratedFromLogEvent"

Reply:
@snakefoot ? Your opinion please :)

Reply:
Changed to GeneratedFromLogEvent